sdk/typescript/test/unit/utils/iota-types.test.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, expect, it } from 'vitest';
+import { normalizeIotaAddress } from '../../../src/utils/iota-types';
+
+describe('normalizeIotaAddress', () => {
+    it('should convert address to lowercase', () => {
+        expect(normalizeIotaAddress('0xABC123')).toBe(
+            '0x0000000000000000000000000000000000000000000000000000000000abc123',
+        );
+    });
+
+    it('should remove whitespaces', () => {
+        expect(normalizeIotaAddress(' 0xab  c1 23 ')).toBe(
+            '0x0000000000000000000000000000000000000000000000000000000000abc123',
+        );
+    });
+
+    it('should add 0x prefix when missing', () => {
+        expect(normalizeIotaAddress('abc123')).toBe(
+            '0x0000000000000000000000000000000000000000000000000000000000abc123',
+        );
+    });
+
+    it('should pad address to correct length', () => {
+        expect(normalizeIotaAddress('0x000123')).toBe(
+            '0x0000000000000000000000000000000000000000000000000000000000000123',
+        );
+    });
+
+    it('should remove 0x prefix and then add it back by default', () => {
+        expect(normalizeIotaAddress('0x123')).toBe(
+            '0x0000000000000000000000000000000000000000000000000000000000000123',
+        );
+    });
+
+    it('should add 0x prefix when forceAdd0x is true', () => {
```

Comment:
Why would it return such an odd address with `0x` at the begning and somewhere mid end? is this a behavior we want?